libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -156,22 +156,23 @@ void ContractLevelChecker::findDuplicateDefinitions(map<string, vector<T>> const
 	}
 }
 
-void ContractLevelChecker::checkAbstractFunctions(ContractDefinition const& _contract)
+void ContractLevelChecker::checkAbstractDefinitions(ContractDefinition const& _contract)
 {
 	// Mapping from name to function definition (exactly one per argument type equality class) and
 	// flag to indicate whether it is fully implemented.
-	using FunTypeAndFlag = std::pair<FunctionTypePointer, bool>;
-	map<string, vector<FunTypeAndFlag>> functions;
+	using ProxyAndFlag = std::pair<OverrideProxy, bool>;
```

Comment:
Since OverrideProxy has `::unimplemented`, I don't think we still need the flag, do we?

Reply:
It's not immediately clear how to get rid of the `bool`. If we remove it, a possible way is to do (see line 176-177) is the following

```cpp
else if (_implemented)
			*it = _overrideProxy
```

But I guess assigning a `bool` would be cheaper than assigning an `OverrideProxy` object every time there is a match. Do you see a better way?

Reply:
I guess that is connected to the other question about using the comparator in the override proxy.